### PR TITLE
Setup/initial setup

### PR DESCRIPTION
**Pull request by andrewvallejo**
This PR is the initial setup for the API. 

## Description
- Add babel, cors, express, and nodemon dependencies. 
- Add hard data stubs {"id":  0, "question":  "", "answers": []}
- Add GET calls for `questions` and `questions/:id`
- Add POST calls submitting answers, which are then added to the data's "answers" array.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor

## Motivation and Context
To give frontend an API to setup fetch calls right  away.

### CODE REVIEW

File: data/questions.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+export const questions = [
+  {
+    "id": 1,
+    "question": "Explain what ARIA and screenreaders are, and how to make a website accessible.",
+    "answers": [],
+  }, {
+    "id": 2,
+    "question": "What is the virtual DOM in React?",
+    "answers": [],
```
Comment (shayan-golafshani) [inline data/questions.js:9]: Nice job, adding a comma at the end. These are good developer empathy habits.

File: server.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const questions = app.locals.questions;
+  response.json({ questions })
+})
+
+app.get('/api/questions/:id', (request, response) => {
+  const questions = app.locals.questions;
+  const { id } = request.params 
+  const queriedQuestion = questions.find(question => question.id === parseInt(id));
+  
+  !queriedQuestion
```
Comment (shayan-golafshani) [inline server.js:25]: Clean use of a ternary

File: server.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    ? response.status(404).send('This question is not found!')
+    : response.status(200).json(queriedQuestion)
+})
+
+app.post('/api/questions/:id', (request, response) => {
+  const { id } = request.params;
+  const { answer } = request.body;
+  
+  !answer && response.status(422)
```
Comment (shayan-golafshani) [inline server.js:34]: Great use of the && operator, but I do think an if-statement here would be a lot more readable.

**Review by shayan-golafshani — APPROVED** [review]
We worked on this together in person, but it's really great. I like the idea of playing with Query Strings and Parameters to make our back-end more robust. Let's delve into that more in future iterations.

File: data/questions.js (hunk 1 above)
Comment (andrewvallejo) [reply data/questions.js:9]: Thanks man, I try!

File: server.js (hunk 3 above)
Comment (andrewvallejo) [reply server.js:34]: Thanks for the tip!

**Pull request merged**